src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexPersistence.cs

```diff
@@ -154,7 +155,25 @@ private void CreatePersistentDictionary(IndexingStatsAggregator indexingStatsAgg
             var defaultDictionaryId = PersistentDictionary.CreateDefault(llt);
             var defaultDictionary = new PersistentDictionary(llt.GetPage(defaultDictionaryId));
 
-            PersistentDictionary.Create(llt, enumerator);
+            var trainedDictionary = PersistentDictionary.Create(llt, enumerator);
+
+            if (defaultDictionaryId != trainedDictionary.DictionaryId)
```

Comment:
Do we need to always create default dictionary? Shouldn't we do it only if we didn't manage to create the one trained on actual indexing data?

Reply:
@redknightlois I assume this is related to changes you made about not erroring the index when training fails. Can you answer?

Reply:
We don't need to create the default dictionary IF we have a trained one. We can just get rid of it.

Reply:
My question is if we can have a code like:

```
if (PersistentDictionary.TryCreate(llt, enumerator) == false)
{
   PersistentDictionary.CreateDefault(llt);
}
```

so instead of creating and deleting the default dict we'll create it only if needed

Reply:
Yes, it could be possible.

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-21591/Corax-Create-default-dictionary-only-if-we-couldnt-create-a-trained-one